src/legacy/core_plugins/kibana/ui_setting_defaults.js

```diff
@@ -67,12 +69,18 @@ export function getUiSettingDefaults() {
         defaultMessage: 'Default route',
       }),
       value: '/app/kibana',
-      validation: {
-        regexString: '^/',
-        message: i18n.translate('kbn.advancedSettings.defaultRoute.defaultRouteValidationMessage', {
-          defaultMessage: 'The route must start with a slash ("/")',
-        }),
-      },
+      schema: schema.string({
```

Comment:
That's not a full equivalent since the old validation was embedded in UI but the new one shown as a notification in a popup. I think it's okay to have the server-side as the source of truth.

Reply:
We lost the `isRelativePath` validation that was specified in `src/legacy/server/http/setup_default_route_provider.ts`. It's unfortunately not sufficient that the `defaultRoute` begins with a `/`; we should also be parsing it to ensure that the path is indeed relative.

For example, in `master`, setting the `defaultRoute` advanced setting to `//example.org` will cause the browser to fallback to the default `/app/kibana` route, since the configured `//example.org` is not a valid path. _However_, in this PR, using `//example.org` will be allowed, and navigating to `http://localhost:5601` will cause Kibana to issue a redirect to `http://example.org`, which we definitely don't want to permit.

Reply:
👍 ok I will put that logic in validation. Just weird that we didn't check this case **in** ui settings declaration `validate`

Reply:
> 👍 ok I will put that logic in validation.

Thanks!

> Just weird that we didn't check this case **in** ui settings declaration `validate`

For sure :) The old `validate` routine was run client-side, and was only put in to improve the UX. We didn't have the infrastructure in place to send down a function for the UI to execute as part of the validation routine, and the server-side infrastructure didn't exist until this PR. Additionally, a client-side validation isn't all that helpful from a security perspective 😄